docs/reference/search/search-your-data/run-knn-search.asciidoc

```diff
@@ -0,0 +1,248 @@
+[[run-knn-search]]
+== Run a k-nearest neighbor (kNN) search
+++++
+<titleabbrev>Run a kNN search</titleabbrev>
+++++
+
+experimental::[]
+
+//tag::knn-def[]
+A _k-nearest neighbor_ (kNN) search finds the _k_ nearest vectors to a query
+vector, as measured by a similarity metric.
+//end::knn-def[]
+
+image::images/kNN/knn_diagram.svg[Approximate kNN search as a two-dimensional graph,align="center"]
+
+Common use cases for kNN include:
+
+* Product recommendations and recommendation engines
+* Similarity search for images or videos
+* Outlier detection and classification
+
+[discrete]
+[[knn-prereqs]]
+=== Prerequisites
+
+* To run a kNN search, you must be able to convert your data into meaningful
+<<dense-vector,`dense_vector`>> values, called embeddings. This typically
```

Comment:
When I think about  embeddings, I think of "word embeddings", which means they are only applicable for texts.  When I think about vectors representing videos or images,  I think more of encoded features.   But may be I have an outdated knowledge, and now we can apply a word "embeddings" even for images. @jtibshirani can comment better on this.

Reply:
May be better just say "vectors" here?

Reply:
Thanks for the feedback here. I think I got a little carried away with the vocab. I revised this to focus on two takeaways:

1) To use kNN, you'll need a way to generate dense vector values for your data.
2) Elastic doesn't currently provide an out-of-the-box method for generating those values.

Changes are in https://github.com/elastic/elasticsearch/pull/80857/commits/29d37f194d53c1da67de4b7376da9939803b3b08. Any other feedback on this bullet is appreciated.